test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/BookkeeperK8sService.java

```diff
@@ -81,6 +81,22 @@ public boolean isRunning() {
                                               t -> new TestFrameworkException(RequestFailed, "Failed to fetch ServiceDetails for bookkeeper", t));
     }
 
+    protected Map<String, Object> patchSpecBuilder(int newInstanceCount) {
+
+        final Map<String, Object> bookkeeperSpec = ImmutableMap.<String, Object>builder()
+                .put("replicas", newInstanceCount)
+                .build();
+
+        return ImmutableMap.<String, Object>builder()
+                .put("apiVersion", "pravega.pravega.io/v1alpha1")
```

Comment:
The same code is getting duplicated across BookkeeperK8sService, PravegaControllerK8sService and PravegaSegmentStoreK8sService. Could you refactor it?

Reply:
done